Source/UI/NewUI/PaymentForm/View/CardNumber/CardNumberViewModel.swift

```diff
@@ -35,13 +35,15 @@ extension CardNumberViewModel: CardNumberViewModelProtocol {
 
     switch cardValidator.validateCompleteness(cardNumber: cardNumber) {
     case .failure:
+        delegate?.update(cardNumber: nil, scheme: .unknown)
         return nil
     // An unknown scheme can also be generated when an eager validation was matched
     //    but the number is not complete.
     // In this case we only update delegate if we have a final result to avoiding
     //    overriding the eager validation with a less informative update
     case .success((let isComplete, let scheme)):
       guard isComplete else {
+        delegate?.update(cardNumber: nil, scheme: .unknown)
```

Comment:
If the number is not complete, does that mean we shouldn't update our record of it? I am unsure why all these return paths are returning `nil` when if looks like a partial number would also work? Would prefer to avoid handling optionality if we don't have to